### DISCUSSION TITLE
Collaborative annotation of texts for collective learning/sensemaking (e.g. for Gebser's Ever Present Origin, or McGilchrist's Matter with Things etc)

### DISCUSSION THREAD

**Discussion by rufuspollock** [life-itself/community#984] [Ideas]
When reading a text (or listening to podcast / video with a transcript) I would like to add commentary (highlight, comments etc) and see that of others plus offer/receive feedback on these so that we have a co-learning and co-sensemaking experience.

- **multiplayer** this is about supporting multiple people annotating
- **async** want to do this in async manner (so outside of a live discussion) so that this can happen without extensive coordination
- **option to limit participants (be private)** want to be able to do this privately (i.e. visibility limited to specific group) *and/or* publicly (for when we want to share results)
  - limiting access may be important for copyright reasons too e.g. difficult to put the full text on the web ...
- **static texts** we're focused here on static texts that don't change e.g. the text of a book or (finalized) transcript of a podcast
- **mobile friendly** (desirable not essential)

This could be relevant for study group or for books e.g. Gebser's Ever Present Origin, or McGilchrist's Matter with Things etc.

## What

Something that looks like this

![image](https://github.com/life-itself/community/assets/180658/648d5309-53c2-4b59-a747-8431418627eb)

## How

MVP: start just getting the text up somewhere and have everyone use hypothes.is. this would be v simple to do.

- could even p/w protect in some way

MVP (option 2): copy a text into google docs and go from there ...

MVP (option 3): (v hacky) use github pull requests ie. create a pull request with full text and then use the pull request for annotation

- think this is a bad idea as v hacky and you lose all the information once you merge.

## Why

Speaking for myself (Rufus)

- I do a lot of annotation and commentary myself ... 
- But it remains in my own personal silo (and often unpublished or not published for a long time)
- Sharing with others would be a major motivation to get stuff out ...
- Moreover, when i share i get feedback that helps clarify what i'm saying ... [^1]
- Conversely, i would benefit from the insights and commentary of others

In short: collective commentary via annotation could lead to better personal and collective sensemaking

- People benefit from sharing as they see the insights and clarifications of others
  - e.g. highlights of text helps you see what parts are important
  - annotation helps clarify and summarize section
- The fact something is shared and that others may look at it motivates sharing
- Constructive critical discussion helps clarify thinking and points of view

[^1]: here's a nice example where this happened from doing public annotation with hypothes.is
     ![image](https://github.com/life-itself/community/assets/180658/03ee9ec0-8d18-4f76-9257-1b3e85b02608)

## Risks / challenges

IME of annotating and summarizing texts have tended to copy from the original text into my own doc and comment there. This allows more detailed commentary, adding emphasis, showing context of text in overall work etc.

However, i have largely been able to replicate the key features I want with hypothesis when i have used it.

Qu: why don't i use hypothesis today? Ans: Because texts aren't online (publicly) to annotate and standard hypothesis only works with online texts (though i note they have integration with various LMS (learning management systems) but we don't have an LMS and anyway not clear on pricing for that  ...)

### Past experience

> [!todo]
>
> incomplete and very partial

Things i've done / tried

- Built my own a couple of times e.g. openshakespeare.org, involved in early hypothes.is
- Used the following apps over the years
  - calibre
  - apple ibook: annoying insertions on copy and paste. No way to export annotations to e.g. markdown from the app
  - readwise: nice reader app (best i've seen actually). But single player only. Plus annotation system is not what i want ultimately.
  - obsidian: actively use this

### Colophon

This is something i've been working on since 2005 and if not earlier 😉. In fact, annotation system in hypothes.is came out of work I did at OKF with @nickstenning. Keep coming back to this need - and don't really understand why we don't have this everywhere yet (or why i can't get my own click and go installation of this ...)

## Footnotes

**Comment by james-baker** [comment]
Hi Rufus, this reminds me of a book club I participated in (it was for McGilchrist's The Master and His Emissary, as a matter of fact) that used Roam to great effect.

We put the full text in, one chapter per Roam page, and then everybody put their own notes/commentary in their own pages -- with the bidirectional linking functionality working really well to be able to read the text, surface and expand other's annotations (through the subtle "reference counts" on any bullet with links), and add your own while never impacting the readability of the shared text.

There was a bit of a tech learning curve (and maybe Obsidian has overtaken Roam now?), but I haven't used Hypothes.is so I can't compare. One more option for consideration though!

**Reply by rufuspollock** [reply]
Nice suggestion. W're just a bit averse to roam given where they've gone. But that's exactly the kind of thing we are looking for.

PS: could you provide a screenshot by any chance?

**Reply by james-baker** [reply]
Ah, I appear to no longer have access to that Roam graph! ☹ Totally get what you mean about Roam as a company.
And of course Obsidian was really designed for single-player... hmm maybe it's improved on "use Dropbox to sync" (or that might be good enough) https://forum.obsidian.md/t/multiplayer-collective-knowledge-building/19852/4

**Reply by rufuspollock** [reply]
I haven't yet found a setup for obsidian that's true "multiplayer" in the way, say, hackmd is for markdown. You can use Visual Studio code but that is a bit geeky.

With the advancing state of [yjs](https://github.com/yjs/yjs) i am hopeful that it won't be long until we have something good. (The problem here is that i think multiplayer for *one* current file is actually quite straightforward ... but doing that for the whole vault is tough and you sort of need the whole vault if you want e.g. autocomplete for file links ...)

**Reply by rufuspollock** [reply]
And lo and behold there is now a yjs based plugin for Obsidian call https://relay.md/

**Comment by farandfew** [comment]
I'd love to have a method for this and a community to work with on it. And, funnily enough, am thinking of it also because of my annotations on TM&HE at the moment.
I have three questions about it:
(1) What's the drawback of the Google Drive option?
(2) Would it be possible to easily export a digest which has only the annotations linked to page numbers, not the actual text? This would allow sharing without copyright worries and would be good for people like me who like making pencil notes in paper copies.
(3) More generally, I feel there's a difference between annotating a book because you're following the author's argument in the author's terms, and when you are looking for lessons for a specific context. Especially as we're talking about some very wide-ranging books here. How might one distinguish between 'we are reading this book to see if we think it's correct' and 'we are reading this book to see if it can help us with what we're doing?'

**Reply by rufuspollock** [reply]
> because of my annotations on TM&HE at the moment.

Great. I published my annotations / summary of that here

https://lifeitself.org/blog/2018/05/01/mcgilchrist-master-and-his-emissary-notes

We also had another piece analysing its impact here:

https://lifeitself.org/blog/2021/06/18/the-fate-of-the-master-and-his-emissary

> (1) What's the drawback of the Google Drive option?

Google Docs is a strong contender. There are a couple of things you can't do ...

- You can't do make the text+annotations easily public (However, this would not be relevant if text were not public - i note i had been thinking of doing this for https://lifeitself.org/sutras which are already public)
- I'm not sure API for google drive comments is that great (though i haven't checked)
- You can't integrate this effort into a broader wiki experience

> (2) Would it be possible to easily export a digest which has only the annotations linked to page numbers, not the actual text? This would allow sharing without copyright worries and would be good for people like me who like making pencil notes in paper copies.

That's a possible requirement but not one i would personally put high on the list given the context here. I mean if you just wnat to do annotation yourself i'd recommend just using one of the many apps for that and then look at that need  in that app.

What i'm getting at is that the focus in this thread was on *collaborative* annotation.

> (3) More generally, I feel there's a difference between annotating a book because you're following the author's argument in the author's terms, and when you are looking for lessons for a specific context. Especially as we're talking about some very wide-ranging books here. How might one distinguish between 'we are reading this book to see if we think it's correct' and 'we are reading this book to see if it can help us with what we're doing?'

Good distinction. I would be happy to be doing both at the same time.

**Reply by farandfew** [reply]
Thanks for pointing me (back) to the notes and essay on TM&HE. I can't seem to download Liam's essay from the institute, btw.
This could be a big rabbit hole for me and, actually, getting stuck in (branching) rabbit holes is my main worry about getting involved in a project of collaborative annotation.
What that makes me think, though, is that the issues that will come up with this are human issues and not necessarily easily foreseen. I'd like to make the suggestion that, if there is enough interest - and enough in a particular text - a trial version could be done on Google Docs and there might be more insight after that into what were the key limitations of that platform. Or, the key limitations of us humans which are revealed by engaging with it :-)
Incidentally the thing with the page numbers - I was thinking it would be useful for me because it would allow me to compare margin annotations in a print copy and then type them out into an online version if I thought they were relevant and helpful. For me, this would take less time, not more, as I can be way too verbose commenting on a digital document. Speaking of which

**Reply by rufuspollock** [reply]
>  I can't seem to download Liam's essay from the institute, btw.

There's an issue there. Flagged in #454 

>  I'd like to make the suggestion that, if there is enough interest - and enough in a particular text - a trial version could be done on Google Docs and there might be more insight after that into what were the key limitations of that platform. Or, the key limitations of us humans which are revealed by engaging with it :-)

Sounds good. I like minimal MVPs.

The next thing would be to pick a text i guess. Then find an epub. Then use pandoc to convert to word format then import to google docs and away you go. would be good to maybe even just start with a part of the book maybe e.g. the first chapter or two.

**Comment by asimong** [comment]
I agree this is a challenging issue, altogether. For commentary on static texts, it seems hard but not extremely hard, and I'd like to see a demo of Roam or Obsidian or preferably an open source alternative with similar functionality for this. On the other hand, for mutable texts, the best I've seen so far is the good old Google Docs, that allows suggestions as well as comments, and keeps records of who has done what; but of course only works per document.

And something that I've not seen anywhere at all, but I would greatly appreciate, is some kind of semantic aid to what the relationship is between the commentary and the text commented on.

Some systems seem to have caught a little of what is wanted here — so for instance LinkedIn allows one to convert a comment on someone else's post to one of your own. That ownership issue seems to me quite important and significant. A while back I created a wish list for wikis to support a living knowledge commons, and I'd guess that much of it is relevant here. I may not need to add that as far as I have researched there is no software that implements all of my ideas, by a long way.

See https://wiki.simongrant.org/doku.php/wiki:requirements-commons

**Reply by rufuspollock** [reply]
> I agree this is a challenging issue, altogether. For commentary on static texts, it seems hard but not extremely hard, and I'd like to see a demo of Roam or Obsidian or preferably an open source alternative with similar functionality for this. On the other hand, for mutable texts, the best I've seen so far is the good old Google Docs, that allows suggestions as well as comments, and keeps records of who has done what; but of course only works per document.

I should have clarified that i'm proposing we **focus only on static texts**. Obsidian won't work for multi-player. I'm not sure what Roam is like here and i'd prefer an open source and/or open format approach if we could (e.g. markdown for text plus e.g. hypothesis for annotation).

> good old Google Docs

It's definitely a strong contender the default "MVP" here i think.

>  A while back I created a wish list for wikis to support a living knowledge commons, and I'd guess that much of it is relevant here. I may not need to add that as far as I have researched there is no software that implements all of my ideas, by a long way.
>
> See https://wiki.simongrant.org/doku.php/wiki:requirements-commons

Nice. This is getting a bit off-topic here ... and very relevant for #992

**Reply by asimong** [reply]
However, I do think we are correct in exploring alternatives to the incumbent services like Google. A couple of vital points… one is that we don't want the sites to break if the service goes down, or starts charging money; two is that we want to ensure that anything we put on an incumbent service is easily extracted in some interoperable format; three is that we want to be supporting the non-capitalist alternatives that are not ultimately driven by advertising revenue.

**Reply by rufuspollock** [reply]
@asimong 👍 and google docs does not really work for those points.
